crates/core/src/client/mod.rs

```diff
@@ -62,7 +62,7 @@ use crate::models::transaction_receipt::StarknetTransactionReceipt as Transactio
 use crate::models::ConversionError;
 
 pub struct KakarotClient<P: Provider + Send + Sync> {
-    starknet_provider: Arc<P>,
+    pub starknet_provider: Arc<P>,
```

Comment:
Any reason for the pub?

Reply:
Ah, this was earlier, when I was dealing with the circular dependency issue, this should not be here. changing

Reply:
resolved [here](https://github.com/kkrt-labs/kakarot-rpc/pull/553/commits/acd1869260b4d43af4642ce5624f8a0bdbb3f592).